tools/autobahn

```diff
@@ -104,6 +106,34 @@ def expand_refvar(module):
     return module
 
 
+def expand_overrides(overrides):
+    for key in overrides:
+        overrides[key] = walk(overrides[key], expand_slot_shorthand)
+        overrides[key] = walk(overrides[key], expand_shortdef)
+    return overrides
+
+
+def process_custom_overrides(tree):
+    if not overrides:
+        return
+    for key in overrides:
+        if key.startswith('root'):
+            apply_custom_override(tree, key)
+
+
+def apply_custom_override(node, name):
+    components = name.split('.')
+    for key in components[1:]:
+        try:
+            if isinstance(node, list):
+                node = node[int(key)]
+            else:
+                node = node['slots'][key]
+        except:
+            return
+    node.update(overrides[name])
```

Comment:
Isn't this going to keep the existing contents of `node` as well as the overridden contents, if they have different keys?

Reply:
Yes, but that is the idea, if you want to fully replace the node, then you need to fully override, but as it is, this allows you to only update certain parts like, e.g. only the properties.